### PR TITLE
chore: bump undici version

### LINKED ISSUE #748: 🐛 BUG: `fetch` behaves differently on Cloudflare Pages and local preview
### What version of `Wrangler` are you using?

0.0.24

### What operating system are you using?

Mac

### Describe the Bug

`fetch` behaves differently on Cloudflare Pages and local preview (`wrangler pages dev`).

#### Steps to reproduce

1. Create `functions/demo.js`
	  ```js
	  // functions/demo.js
	  export const onRequestGet = async () => {
	    try {
	      await fetch('https://wordpress.org/')
	      return new Response('Fetch finished')
	    } catch (err) {
	      return new Response(JSON.stringify({
	        name: err.name,
	        message: err.message,
	        cause: err.cause,
	        stack: err.stack
	      }))
	    }
	  }
	  ```
1. Preview via `wrangler pages dev`. (Files in `./out` folder don't matter)
	```bash
	npx wrangler pages dev ./out
	````
1. Get the `/demo` page from local preview
    ```sh-session
    $ curl http://localhost:8788/demo
	{"name":"TypeError","message":"fetch failed","cause":{"code":"ERR_INVALID_CHAR"},"stack":"TypeError: fetch failed\n    at Object.processResponse (/Users/me/Code/my-demo/node_modules/undici/lib/fetch/index.js:183:23)\n    at Fetch.fetchFinale (/Users/me/Code/my-demo/node_modules/undici/lib/fetch/index.js:937:17)\n    at Fetch.mainFetch (/Users/me/Code/my-demo/node_modules/undici/lib/fetch/index.js:744:17)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)"}
    ```
1. Deploy to Cloudflare Pages
1. Get the `/demo` page from Cloudflare Pages
    ```sh-session
    $ curl https://xxx.my-page.pages.dev/demo
    Fetch finished
    ````
  
  #### Expected Behaviour 
  
  Steps 3 and 5 yield the same result.

### PR DESCRIPTION
**Pull request by cameron-robey**
chore: bump undici and increase minimum node version to 16.13

- We bump undici to version to 5.9.1 to patch some security vulnerabilities in previous versions
- This requires bumping the minimum node version to >= 16.8 so we update the minimum to the LTS 16.13

Fixes https://github.com/cloudflare/wrangler2/issues/1684
Fixes https://github.com/cloudflare/wrangler2/issues/1679
Fixes #748

### CODE REVIEW

**Comment by cameron-robey** [comment]
Seems to be some behaviour change to formData so some tests failing

**Comment by cameron-robey** [comment]
Might want to wait until the next version of Miniflare, as a similar bump was done [here](https://github.com/cloudflare/miniflare/pull/333) and we can get it all bumped in one

File: packages/wrangler/src/__tests__/test-old-node-version.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// this test has to be run with a version of node.js older than 16.7 to pass
+// this test has to be run with a version of node.js older than 13 to pass
```
Comment (rozenmd) [inline packages/wrangler/src/__tests__/test-old-node-version.js:1]:
```suggestion
// this test has to be run with a version of node.js older than 16.13 to pass
```

**Comment by cameron-robey** [comment]
Latest commit fixes https://github.com/cloudflare/wrangler2/issues/1679

**Comment by Skye-31** [comment]
This should also fix #748

**Review by rozenmd — APPROVED** [review]

**Comment by cameron-robey** [comment]
Assuming based on https://github.com/cloudflare/miniflare/pull/333#issuecomment-1222194515, that @threepointone might have some comments...

File: .changeset/young-rabbits-train.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+"node-app-pages": patch
+"pages-functions-app": patch
+"pages-workerjs-app": patch
+"remix-pages-app": patch
```
Comment (threepointone) [inline .changeset/young-rabbits-train.md:5]: let's not add fixtures in changesets
```suggestion
```

**Review by threepointone — APPROVED** [review]
one edit for the changeset, but otherwise looks good

File: .changeset/young-rabbits-train.md (hunk 2 above)
Comment (cameron-robey) [reply .changeset/young-rabbits-train.md]: resolved: https://github.com/cloudflare/wrangler2/pull/1691/commits/fb1d32f80224fd1f9c2d1e6d554fe19939cb6f58

**Pull request merged**